providers/rethinkdb.js

```diff
@@ -1,58 +1,56 @@
-// Copyright (c) 2017-2018 dirigeants. All rights reserved. MIT license.
+// Copyright (c) 2017-2019 dirigeants. All rights reserved. MIT license.
 const { Provider, util: { mergeDefault, chunk } } = require('klasa');
-const rethink = require('rethinkdbdash');
+const { r } = require('rethinkdb-ts'); // eslint-disable-line id-length
 
 module.exports = class extends Provider {
 
 	constructor(...args) {
 		super(...args);
-		this.db = rethink(mergeDefault({
-			db: 'test',
-			user: 'admin',
-			password: '',
-			silent: false
-		}, this.client.options.providers.rethinkdb));
+
+		this.db = r;
+		this.pool = null;
 	}
 
 	async init() {
-		const { db } = this.db._poolMaster._options;
-		await this.db.branch(this.db.dbList().contains(db), null, this.db.dbCreate(db)).run();
-	}
+		const options = mergeDefault({
+			db: 'test',
+			silent: false
+		}, this.client.options.providers.rethinkdb);
 
-	get exec() {
-		return this.db;
+		this.pool = await r.connectPool(options);
+		await this.db.branch(this.db.dbList().contains(options.db), null, this.db.dbCreate(options.db)).run();
 	}
 
 	async ping() {
 		const now = Date.now();
-		return await this.db.now() - now;
+		return (await this.db.now().run()).getTime() - now;
 	}
 
-	async shutdown() {
-		return this.db.getPoolMaster().drain();
+	shutdown() {
+		return this.pool.drain();
 	}
 
 	/* Table methods */
 
-	async hasTable(table) {
+	hasTable(table) {
 		return this.db.tableList().contains(table).run();
 	}
 
-	async createTable(table) {
+	createTable(table) {
 		return this.db.tableCreate(table).run();
 	}
 
-	async deleteTable(table) {
+	deleteTable(table) {
 		return this.db.tableDrop(table).run();
 	}
 
-	async sync(table) {
+	sync(table) {
 		return this.db.table(table).sync().run();
 	}
 
 	/* Document methods */
 
-	async getAll(table, entries = []) {
+	async getAll(table, entries) {
```

Comment:
Any reason to remove the initializer? It will cause a TypeError on `if (entries.length)`